src/ReplaySubject.ts

```diff
@@ -90,12 +116,13 @@ export class ReplaySubject<T> extends Subject<T> {
       spliceCount = Math.max(spliceCount, eventsCount - _bufferSize);
     }
 
-    if (spliceCount > 0) {
-      _events.splice(0, spliceCount);
+    for (let i = 0; i < spliceCount; i++) {
+      _events.shift();
     }
```

Comment:
I'm actually shocked this would be faster.  this section was O(1) now it's O(n).